src/Apps/Order/Routes/__tests__/Shipping.jest.tsx

```diff
@@ -152,68 +121,64 @@ const emptyTestMe: ShippingTestQuery$rawResponse["me"] = {
   },
 }
 
-const testMe: ShippingTestQuery$rawResponse["me"] = {
-  name: "Test Name",
-  email: "test@gmail.com",
-  id: "4321",
-  location: {
-    id: "123",
-    country: "United States",
-  },
-  addressConnection: {
-    totalCount: 0,
-    edges: [
-      {
-        node: {
-          __typename: "UserAddress",
-          internalID: "1",
-          addressLine1: "1 Main St",
-          addressLine2: "",
-          addressLine3: "",
-          city: "Madrid",
-          country: "ES",
-          isDefault: false,
-          name: "Test Name",
-          phoneNumber: "555-555-5555",
-          postalCode: "28001",
-          region: "",
-          id: "addressID1",
+const meWithAddresses: ShippingTestQuery$rawResponse["me"] = Object.assign(
+  {},
+  meWithoutAddress,
+  {
+    addressConnection: {
+      totalCount: 0,
+      edges: [
+        {
+          node: {
+            __typename: "UserAddress",
+            internalID: "1",
+            addressLine1: "1 Main St",
+            addressLine2: "",
+            addressLine3: "",
+            city: "Madrid",
+            country: "ES",
+            isDefault: false,
+            name: "Test Name",
+            phoneNumber: "555-555-5555",
+            postalCode: "28001",
+            region: "",
+            id: "addressID1",
+          },
+          cursor: "aaa",
         },
-        cursor: "aaa",
-      },
-      {
-        node: {
-          __typename: "UserAddress",
-          internalID: "2",
-          addressLine1: "401 Broadway",
-          addressLine2: "Floor 25",
-          addressLine3: "",
-          city: "New York",
-          country: "US",
-          isDefault: true,
-          name: "Test Name",
-          phoneNumber: "422-424-4242",
-          postalCode: "10013",
-          region: "NY",
-          id: "addressID2",
+        {
+          node: {
+            __typename: "UserAddress",
+            internalID: "2",
+            addressLine1: "401 Broadway",
+            addressLine2: "Floor 25",
+            addressLine3: "",
+            city: "New York",
+            country: "US",
+            isDefault: true,
+            name: "Test Name",
+            phoneNumber: "422-424-4242",
+            postalCode: "10013",
+            region: "NY",
+            id: "addressID2",
+          },
+          cursor: "aaa",
         },
-        cursor: "aaa",
-      },
-    ],
-    pageInfo,
-  },
-}
-
-class ShippingTestPage extends OrderAppTestPage {
-  async selectPickupOption() {
-    this.find(`[data-test="pickupOption"]`).last().simulate("click")
-    await this.update()
+      ],
+      pageInfo,
+    },
   }
+)
+
+const saveAndContinue = async () => {
+  await userEvent.click(screen.getByText("Save and Continue"))
+  await flushPromiseQueue()
```

Comment:
Ideally when writing tests I shouldn't need to know if/where I should wait, which is more like implementation details. Wrapping it in a helper, like this, helps encapsulate things but I still need to wait explicitly here and there. Perhaps it's inevitable in unit tests. An alternative would be like Cypress that the tests internally _waits_ for some UI updates.

Reply:
can you use something like `await waitFor(() => {expect(screen.getByText("My success message").toBeInTheDocument())})`? This lets you define an expectation as part of your action, which could be good if the action has a consistent result.

Reply:
Another option since this is just a click helper could be to wait for the loading indicator to go away, since that should always happen.

Reply:
Yeah, but my question was to not worry about needing to wait or not for each action. I see the [`findBy`](https://testing-library.com/docs/queries/about#types-of-queries) queries perform retries and that's more similar to Cypress, and potentially can be an alternative.